fsspec/implementations/cached.py

```diff
@@ -400,7 +400,21 @@ def _open(
                 )
         else:
             detail["blocksize"] = f.blocksize
-        f.cache = MMapCache(f.blocksize, f._fetch_range, f.size, fn, blocks)
+
+        def _fetch_ranges(ranges):
+            return self.fs.cat_ranges(
+                [path] * len(ranges),
+                [r[0] for r in ranges],
+                [r[1] for r in ranges],
+                **kwargs,
+            )
+
+        multi_fetcher = (
+            None if not self.fs.async_impl or self.compression else _fetch_ranges
```

Comment:
It might still be useful for a sync filesystem, since cat_ranges might combine some adjacent reads.